### PR TITLE
Channel assets added   #65

### PR DESCRIPTION
**Pull request by ZiyodullohDilshodov**

### CODE REVIEW

File: src/Shamsheer.Messenger.Api/Controllers/ChannelAssets/ChannelAssetsController.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    {
+        private readonly IChannelAssetService _channelAssetService;
+
+        public ChannelAssetsController(IChannelAssetService channelAssetService)
+        {
+            _channelAssetService = channelAssetService;
+        }
+
+
+        [HttpPost]
```
Comment (dotnetgirl) [inline src/Shamsheer.Messenger.Api/Controllers/ChannelAssets/ChannelAssetsController.cs]: Shu yerda {channel-id} kelishi kerak

File: src/Shamsheer.Messenger.Api/Controllers/ChannelAssets/ChannelAssetsController.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        public async Task<IActionResult> PostAsync(IFormFile file)
```
Comment (dotnetgirl) [inline src/Shamsheer.Messenger.Api/Controllers/ChannelAssets/ChannelAssetsController.cs]: channel-id'ni parametr qilib bervorishiz kerak

File: src/Shamsheer.Messenger.Api/Controllers/ChannelAssets/ChannelAssetsController.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            => Ok(await _channelAssetService.CreateAsync(file));
+
+
+        [HttpGet("{channel-id}")]
+        public async Task<IActionResult> GetAllAsync([FromQuery] PaginationParams @params, [FromRoute] long channelId)
```
Comment (dotnetgirl) [inline src/Shamsheer.Messenger.Api/Controllers/ChannelAssets/ChannelAssetsController.cs]: Hozir berayotgan parametringizni "channel-id" ko'rinishida kelishini aytib ketmasez ikki martta kiritishga to'g'ri kelib qoladi.

File: src/Shamsheer.Messenger.Api/Controllers/ChannelAssets/ChannelAssetsController.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        => Ok(await _channelAssetService.RetriveAllAsync(channelId, @params));
+
+
+        [HttpGet("{channel-id}/{id}")]
+        public async Task<IActionResult> GetAsync([FromRoute(Name = "channelId")] long channelId, [FromRoute(Name = "id")] long id)
```
Comment (dotnetgirl) [inline src/Shamsheer.Messenger.Api/Controllers/ChannelAssets/ChannelAssetsController.cs]: ikkita channel-id so'raydi.

File: src/Shamsheer.Service/Interfaces/ChannelAssets/IChannelAssetService.cs [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+using Shamsheer.Service.DTOs.ChannelAssets;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+
+namespace Shamsheer.Service.Interfaces.ChannelAssets;
+
+public  interface IChannelAssetService
+{
+    public Task<bool> RemoveAsync(long channelId, long id);
+    public Task<ChannelAssetForResultDto> CreateAsync(IFormFile file);
```
Comment (dotnetgirl) [inline src/Shamsheer.Service/Interfaces/ChannelAssets/IChannelAssetService.cs]: Create methodini optimallashtirganmiz, ya'ni o'ziga yana bitta parametr oladi. long channelId

File: src/Shamsheer.Service/Services/ChannelAssets/ChannelAssetService.cs [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+
+
+
+    public async Task<bool> RemoveAsync(long channelId, long id)
+    {
+        var channel = await _channelRepository.SelectAll()
+            .Where(c => c.Id == channelId)
+            .FirstOrDefaultAsync();
+
+        if (channel == null)
```
Comment (dotnetgirl) [inline src/Shamsheer.Service/Services/ChannelAssets/ChannelAssetService.cs]: channel is null deb tekshirganiz yaxshiroq.

File: src/Shamsheer.Service/Services/ChannelAssets/ChannelAssetService.cs [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+            throw new ShamsheerException(404, "Channel is not found");
+
+
+        var channelAsste = await _channelRepository.SelectAll()
+            .Where(c => c.Id == id)
+            .FirstOrDefaultAsync();
+
+        if (channelAsste == null)
+            throw new ShamsheerException(404, "Channel is not found");
```
Comment (dotnetgirl) [inline src/Shamsheer.Service/Services/ChannelAssets/ChannelAssetService.cs]: Channel Asset is not found*

File: src/Shamsheer.Service/Services/ChannelAssets/ChannelAssetService.cs [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+
+        if (channel == null)
+            throw new ShamsheerException(404, "Channel is not found");
+
+
+        var channelAsste = await _channelRepository.SelectAll()
+            .Where(c => c.Id == id)
+            .FirstOrDefaultAsync();
+
+        if (channelAsste == null)
```
Comment (dotnetgirl) [inline src/Shamsheer.Service/Services/ChannelAssets/ChannelAssetService.cs]: is null

File: src/Shamsheer.Service/Services/ChannelAssets/ChannelAssetService.cs [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+            throw new ShamsheerException(404, "Channel is not found");
+
+
+        await _channelAssetRepository.DeleteAsync(channelId);
+
+        return true;
+    }
+
+    public async Task<ChannelAssetForResultDto> CreateAsync(IFormFile file)
```
Comment (dotnetgirl) [inline src/Shamsheer.Service/Services/ChannelAssets/ChannelAssetService.cs]: Bu method o'ziga long channelId oladigan bo'lgan, jwt token'dan yechib olish o'rniga

File: src/Shamsheer.Service/Services/ChannelAssets/ChannelAssetService.cs [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+            .FirstOrDefaultAsync();
+
+        if (channel == null)
+            throw new ShamsheerException(404, "Channel is not found");
+            var channelAsset = await _channelAssetRepository.SelectAll()
+              .Where(c => c.Id == id)
+              .FirstOrDefaultAsync();
+
+            if (channelAsset == null)
+                throw new ShamsheerException(404, "Channel is not found");
```
Comment (dotnetgirl) [inline src/Shamsheer.Service/Services/ChannelAssets/ChannelAssetService.cs]: Channel Asset is not found*

File: src/Shamsheer.Service/Services/ChannelAssets/ChannelAssetService.cs [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+        return _mapper.Map<IEnumerable<ChannelAssetForResultDto>>(asset);
+    }
+
+    public async Task<ChannelAssetForResultDto> RetriveByIdAsync(long channelId, long id)
+    {
+        var channel = await _channelRepository.SelectAll()
+            .Where(c => c.Id == channelId)
+            .FirstOrDefaultAsync();
+
+        if (channel == null)
```
Comment (dotnetgirl) [inline src/Shamsheer.Service/Services/ChannelAssets/ChannelAssetService.cs]: is null

File: src/Shamsheer.Service/Services/ChannelAssets/ChannelAssetService.cs [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+        return _mapper.Map<ChannelAssetForResultDto>(result);
+    }
+
+    public async Task<IEnumerable<ChannelAssetForResultDto>> RetriveAllAsync(long channelId, PaginationParams @params)
+    {
+        var channel = await _channelRepository.SelectAll()
+            .Where(c => c.Id == channelId)
+        .FirstOrDefaultAsync();
+
+        if (channel == null)
```
Comment (dotnetgirl) [inline src/Shamsheer.Service/Services/ChannelAssets/ChannelAssetService.cs]: is null

File: src/Shamsheer.Service/Services/ChannelAssets/ChannelAssetService.cs [hunk 13]
```diff
@@ -0,0 +1,135 @@
...
+
+        await _channelAssetRepository.DeleteAsync(channelId);
+
+        return true;
+    }
+
+    public async Task<ChannelAssetForResultDto> CreateAsync(IFormFile file)
+    {
+        //TODO:LOGIC
+        long channelId = 1; //<= jwt token
```
Comment (dotnetgirl) [inline src/Shamsheer.Service/Services/ChannelAssets/ChannelAssetService.cs]: bu qism kerak emas endi.

Comment (ZiyodullohDilshodov) [reply src/Shamsheer.Service/Services/ChannelAssets/ChannelAssetService.cs]: ok

File: src/Shamsheer.Service/Services/ChannelAssets/ChannelAssetService.cs (hunk 12 above)
Comment (ZiyodullohDilshodov) [reply src/Shamsheer.Service/Services/ChannelAssets/ChannelAssetService.cs]: ok

File: src/Shamsheer.Service/Services/ChannelAssets/ChannelAssetService.cs (hunk 11 above)
Comment (ZiyodullohDilshodov) [reply src/Shamsheer.Service/Services/ChannelAssets/ChannelAssetService.cs]: ok

File: src/Shamsheer.Service/Services/ChannelAssets/ChannelAssetService.cs (hunk 10 above)
Comment (ZiyodullohDilshodov) [reply src/Shamsheer.Service/Services/ChannelAssets/ChannelAssetService.cs]: ok

File: src/Shamsheer.Service/Services/ChannelAssets/ChannelAssetService.cs (hunk 9 above)
Comment (ZiyodullohDilshodov) [reply src/Shamsheer.Service/Services/ChannelAssets/ChannelAssetService.cs]: ok

File: src/Shamsheer.Service/Interfaces/ChannelAssets/IChannelAssetService.cs (hunk 5 above)
Comment (ZiyodullohDilshodov) [reply src/Shamsheer.Service/Interfaces/ChannelAssets/IChannelAssetService.cs]: ok

File: src/Shamsheer.Messenger.Api/Controllers/ChannelAssets/ChannelAssetsController.cs (hunk 4 above)
Comment (ZiyodullohDilshodov) [reply src/Shamsheer.Messenger.Api/Controllers/ChannelAssets/ChannelAssetsController.cs]: ok

File: src/Shamsheer.Messenger.Api/Controllers/ChannelAssets/ChannelAssetsController.cs (hunk 3 above)
Comment (ZiyodullohDilshodov) [reply src/Shamsheer.Messenger.Api/Controllers/ChannelAssets/ChannelAssetsController.cs]: ok

File: src/Shamsheer.Messenger.Api/Controllers/ChannelAssets/ChannelAssetsController.cs (hunk 2 above)
Comment (ZiyodullohDilshodov) [reply src/Shamsheer.Messenger.Api/Controllers/ChannelAssets/ChannelAssetsController.cs]: ok

File: src/Shamsheer.Messenger.Api/Controllers/ChannelAssets/ChannelAssetsController.cs (hunk 1 above)
Comment (ZiyodullohDilshodov) [reply src/Shamsheer.Messenger.Api/Controllers/ChannelAssets/ChannelAssetsController.cs]: ok

File: src/Shamsheer.Messenger.Api/Controllers/ChannelAssets/ChannelAssetsController.cs [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+        private readonly IChannelAssetService _channelAssetService;
+
+        public ChannelAssetsController(IChannelAssetService channelAssetService)
+        {
+            _channelAssetService = channelAssetService;
+        }
+
+
+        [HttpPost("{channel-id}")]
+        public async Task<IActionResult> PostAsync(IFormFile file,[FromRoute(Name = "channel-id")] long channelId)
```
Comment (dotnetgirl) [inline src/Shamsheer.Messenger.Api/Controllers/ChannelAssets/ChannelAssetsController.cs]: birinchi id kelishi kerak, keyin iform file.

File: src/Shamsheer.Messenger.Api/Controllers/ChannelAssets/ChannelAssetsController.cs [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+            => Ok(await _channelAssetService.CreateAsync(file , channelId));
```
Comment (dotnetgirl) [inline src/Shamsheer.Messenger.Api/Controllers/ChannelAssets/ChannelAssetsController.cs]: channelId, file

File: src/Shamsheer.Service/Interfaces/ChannelAssets/IChannelAssetService.cs [hunk 16]
```diff
@@ -0,0 +1,19 @@
...
+using Shamsheer.Service.DTOs.ChannelAssets;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+
+namespace Shamsheer.Service.Interfaces.ChannelAssets;
+
+public  interface IChannelAssetService
+{
+    public Task<bool> RemoveAsync(long channelId, long id);
+    public Task<ChannelAssetForResultDto> CreateAsync(IFormFile file , long channelId);
```
Comment (dotnetgirl) [inline src/Shamsheer.Service/Interfaces/ChannelAssets/IChannelAssetService.cs]: long channelId, IFormFile file

File: src/Shamsheer.Service/Services/ChannelAssets/ChannelAssetService.cs [hunk 17]
```diff
@@ -0,0 +1,133 @@
...
+
+        await _channelAssetRepository.DeleteAsync(channelId);
+
+        return true;
+    }
+
+    public async Task<ChannelAssetForResultDto> CreateAsync(IFormFile file , long channelId)
+    {
+        var channel = await _channelRepository.SelectAll()
+            .Where(x => x.Id == channelId)
```
Comment (dotnetgirl) [inline src/Shamsheer.Service/Services/ChannelAssets/ChannelAssetService.cs]: nimaga bu x?

File: src/Shamsheer.Service/Services/ChannelAssets/ChannelAssetService.cs [hunk 18]
```diff
@@ -0,0 +1,133 @@
...
+        if (channelAsste is null)
+            throw new ShamsheerException(404, "ChannelAsset is not found");
+
+
+        await _channelAssetRepository.DeleteAsync(channelId);
+
+        return true;
+    }
+
+    public async Task<ChannelAssetForResultDto> CreateAsync(IFormFile file , long channelId)
```
Comment (dotnetgirl) [inline src/Shamsheer.Service/Services/ChannelAssets/ChannelAssetService.cs]: birinchi channelId olishi kerak parametr sifatida

File: src/Shamsheer.Messenger.Api/Controllers/ChannelAssets/ChannelAssetsController.cs (hunk 14 above)
Comment (ZiyodullohDilshodov) [reply src/Shamsheer.Messenger.Api/Controllers/ChannelAssets/ChannelAssetsController.cs]: Sabab

File: src/Shamsheer.Messenger.Api/Controllers/ChannelAssets/ChannelAssetsController.cs (hunk 15 above)
Comment (ZiyodullohDilshodov) [reply src/Shamsheer.Messenger.Api/Controllers/ChannelAssets/ChannelAssetsController.cs]: Sabab

File: src/Shamsheer.Service/Interfaces/ChannelAssets/IChannelAssetService.cs (hunk 16 above)
Comment (ZiyodullohDilshodov) [reply src/Shamsheer.Service/Interfaces/ChannelAssets/IChannelAssetService.cs]: nega

File: src/Shamsheer.Service/Services/ChannelAssets/ChannelAssetService.cs (hunk 18 above)
Comment (ZiyodullohDilshodov) [reply src/Shamsheer.Service/Services/ChannelAssets/ChannelAssetService.cs]: sabab

File: src/Shamsheer.Service/Services/ChannelAssets/ChannelAssetService.cs (hunk 17 above)
Comment (ZiyodullohDilshodov) [reply src/Shamsheer.Service/Services/ChannelAssets/ChannelAssetService.cs]: x  ligi uchun x )

**Pull request merged**